documentation/guides/migrating-from-v3-to-v4.md

```diff
@@ -0,0 +1,417 @@
+
+### WithContext <a name="polaris-withcontext"></a>
+
+`WithContext` as an internal abstraction used for class components. Use hooks or contextType instead.
+
+```jsx
+// old
+function Test({context}) {
+  console.log(context);
+  return null;
+}
+
+const TestWithContext = withContext(Consumer)(Test);
+
+// new with hooks
+function Test() {
+  const testContext = useContext(TestContext);
+  return <h1>{testContext}</h1>;
+}
+
+// new with contextType
+const TestContext = createContext('Polaris');
+
+class Test extends Component {
+  static contextType = TestContext;
+
+  render() {
+    const testContext = this.context;
+    return <h1>{testContext}</h1>;
+  }
+}
+```
+
+### WithRef <a name="polaris-withref"></a>
+
+`WithRef` was an internal component used to place refs on components wrapped in higher-order components. Use functional components instead.
+
+```jsx
+// old
+class Input extends React.PureComponent {
+  inputRef = createRef();
+
+  focus = () => {
+    inputRef.current.focus();
+  };
+
+  render() {
+    return <input ref={inputRef} {...props} />;
+  }
+}
+
+const WrappedInput = compose(
+  hocOne,
+  hocTwo,
+  hocX,
+  withRef,
+)(Input);
+
+class App extends React.PureComponent {
+  input = createRef();
+
+  componentDidMount() {
+    input.current.focus();
+  }
+
+  render() {
+    return (
+      <ReffableInput
+        ref={reffableInput}
+        onFocus={() => console.log('focused')}
+      />
+    );
+  }
+}
+
+// new with hooks
+function Input(props, ref) {
+  const inputRef = useRef();
+  useImperativeHandle(ref, () => ({
+    focus: () => {
+      inputRef.current.focus();
+    },
+  }));
+  return <input ref={inputRef} {...props} />;
+}
+const ReffableInput = forwardRef(Input);
+
+function App() {
+  const reffableInput = useRef();
+
+  useEffect(() => {
+    reffableInput.current.focus();
+  }, []);
+
+  return (
+    <ReffableInput ref={reffableInput} onFocus={() => console.log('focused')} />
+  );
+}
+```
+
+### Autocomplete.ComboBox.TextField <a name="polaris-autocompletecomboboxtextfield"></a>
+
+Use `Autocomplete.TextField` instead.
+
+```jsx
+...
+const textFieldMarkup = (
+  <Autocomplete.TextField
+    label="Example"
+  />
+);
+...
+```
+
+### Autocomplete.ComboBox.OptionList <a name="polaris-autocompletecomboboxoptionlist"></a>
+
+Use `OptionList` instead.
+
+```jsx
+...
+const optionListMarkup = (
+  <OptionList
+    onChange={() => {}}
+    selected={[]}
+  />
+);
+...
+```
+
+### Navigation.UserMenu <a name="polaris-navigationusermenu"></a>
+
+Use `TopBar.UserMenu` instead
+
+```jsx
+const userMenuMarkup = (
+  <TopBar.UserMenu
+    actions={[]}
+    name="Andrew"
+    initials="AM"
+    open
+    onToggle={() => {}}
+  />
+);
+```
+
+### Modal.Dialog/Tabs.Panel <a name="polaris-dialog-panel"></a>
+
+`Dialog` and `Panel` are subcomponents that are used by Polaris to build the respective component and are not intended to be used outside of their component.
```

Comment:
Was there a change here or are we simply using this to tell people not to use these outside these components?